step-plans/step-plans-core/src/main/java/step/core/artefacts/handlers/ArtefactHandler.java

```diff
@@ -128,11 +128,18 @@ public void createReportSkeleton(ReportNode parentReportNode, ARTEFACT artefact,
 	protected abstract void createReportSkeleton_(REPORT_NODE parentNode, ARTEFACT testArtefact);
 	
 	public ReportNode execute(REPORT_NODE parentReportNode, ARTEFACT artefact, Map<String, Object> newVariables) {
-		if (artefact.getSkipNode().get()) {
-			REPORT_NODE reportNode = createReportNode(parentReportNode, artefact);
-			reportNode.setStatus(ReportNodeStatus.SKIPPED);
-			reportNode.setExecutionTime(System.currentTimeMillis());
-			return reportNode;
+		//Handle skip node at the beginning to avoid unnecessary processing and persistence
+		try {
+			//We still need to evaluate expressions since it can be defined dynamically in the UI
+			dynamicBeanResolver.evaluate(artefact, getBindings());

```

Comment:
@david-stephan Having the check at this places would ignore the settings TEC_EXECUTION_REPORTNODES_PERSISTONLYNONPASSED. I would therefore recommend to have the check to the place where we handle nodes skipped by filters (see following comment)